specs/core/1_custody-game.md

```diff
@@ -536,252 +352,93 @@ def process_early_derived_secret_reveal(state: BeaconState, reveal: EarlyDerived
+    
+    # Compute the custody bit
+    computed_custody_bit = compute_custody_bit(custody_slashing.malefactor_key, custody_slashing.data)
+    
+    # Verify the claim
+    if claimed_custody_bit != computed_custody_bit:
+        # Slash the malefactor, reward the other committee members
+        slash_validator(state, custody_slashing.malefactor_index)
+        others_count = len(committee) - 1
+        whistleblower_reward = Gwei(malefactor.effective_balance // WHISTLEBLOWER_REWARD_QUOTIENT // others_count)
+        for attester_index in attesters:
+            if attester_index != custody_slashing.malefactor_index:
+                increase_balance(state, attester_index, whistleblower_reward)
+        # No special whisteblower reward: it is expected to be an attester. Others are free to slash too however. 
+    else:
+        # The claim was false, the custody bit was correct. Slash the whistleblower that induced this work.
+        slash_validator(state, custody_slashing.whistleblower_index)
 ```
 
-```python
-def process_bit_challenge_response(state: BeaconState,
-                                   response: CustodyResponse,
-                                   challenge: CustodyBitChallengeRecord) -> None:
-    # Verify chunk index
-    assert response.chunk_index < challenge.chunk_count
-    # Verify responder has not been slashed
-    responder = state.validators[challenge.responder_index]
-    assert not responder.slashed
-    # Verify the chunk matches the crosslink data root
-    assert is_valid_merkle_branch(
-        leaf=hash_tree_root(response.chunk),
-        branch=response.data_branch,
-        depth=ceillog2(challenge.chunk_count),
-        index=response.chunk_index,
-        root=challenge.data_root,
-    )
-    # Verify the chunk bit leaf matches the challenge data
-    assert is_valid_merkle_branch_with_mixin(
-        leaf=hash_tree_root(response.chunk_bits_leaf),
-        branch=response.chunk_bits_branch,
-        depth=ceillog2(MAX_CUSTODY_CHUNKS // 256),
-        index=response.chunk_index // 256,
-        root=challenge.chunk_bits_merkle_root,
-        mixin=challenge.chunk_count,
-    )
-    # Verify the chunk bit does not match the challenge chunk bit
-    assert (get_custody_chunk_bit(challenge.responder_key, response.chunk)
-            != response.chunk_bits_leaf[response.chunk_index % 256])
-    # Clear the challenge
-    records = state.custody_bit_challenge_records
-    records[records.index(challenge)] = CustodyBitChallengeRecord()
-    # Slash challenger
-    slash_validator(state, challenge.challenger_index, challenge.responder_index)
-```
 
 ## Per-epoch processing
 
-### Handling of custody-related deadlines
+### Handling of reveal deadlines
 
-Run `process_reveal_deadlines(state)` immediately after `process_registry_updates(state)`:
+Run `process_reveal_deadlines(state)` after `process_registry_updates(state)`:
 
 ```python
-# begin insert @process_reveal_deadlines
-    process_reveal_deadlines(state)
```

Comment:
Don't we still need this?

Reply:
We don't use the previous comment-based code generation method anymore. It was brittle and difficult to follow for clients and other contributors.

Instead we override functions and objects explicitly and fully in subsequent phases. For example in phase 1, we fully redefine `process_epoch` explicitly writing the previous and new calls

https://github.com/ethereum/eth2.0-specs/pull/1504/files#diff-fea137c7b15f508b0dce4aeb83f9e98cR759-R768